lib/plugin/selenoid.js

```diff
@@ -0,0 +1,207 @@
+    options.capabilities['selenoid:options'].videoName = `${test.title}.mp4`;
+    options.capabilities['selenoid:options'].logName = `${test.title}.log`;
+    WebDriver._setConfig(options);
+  });
+};
+
+const replaceScriptConfig = (config) => {
+  for (const key of Object.keys(config)) {
+    dockerCreateScript = dockerCreateScript.replace(new RegExp(`\\$${key}\\$`, 'g'), config[key]);
+  }
+  dockerStartScript = dockerStartScript.replace('$name$', config.name);
+  dockerStopScript = dockerStopScript.replace('$name$', config.name);
+  seleniumUrl = seleniumUrl.replace('$port$', config.port);
+};
+
+
+/**
+ * # Selenoid
+ *
+ * Selenoid plugin with for recording video recording and logs
+ *
+ * ## Prerequisite
+ *
+ *  - Docker
+ *
+ * ## How to ?
+ *
+ * ### 1. Add browsers.json
+ * Add browsers.json parallel to codecept conf location. [Refer here](https://aerokube.com/selenoid/latest/#_prepare_configuration) to know more about browsers.json
+ *
+ * **You can download the sample json from example.**
+ *
+ * ### 2. Add plugin configuration in codecept
+ *
+ *  Add plugin config to codecept conf.
+ * ```js
+ * plugins: {
+ *     selenoid: {
+ *       enabled: true,
+ *       deletePassed: true,
+ *       autoCreate: true,
+ *       autoStart: true,
+ *       sessionTimeout: '30m',
+ *       enableVideo: true,
+ *       enableLog: true,
+ *     },
+ *   }
+ * ```
+ * #### Options:
+ * | Param | Description |
+ * |--|--|
+ * | name | Name of the container (default : selenoid) |
+ * | port | Port of selenium server (default : 4444) |
+ * | deletePassed | Delete video and logs of passed tests (default : true)|
+ * | autoCreate | Will automatically create container (Linux only) (default : true)|
+ * | autoStart | If disabled start the container manually before running tests (default : true)|
+ * | enableVideo | Enable video recording (`video` folder of output)|
+ * | enableLog | Enable video recording (`logs` folder of output)|
+ * | additionalParams | example: `additionalParams: '--env TEST=test'` [Refer here](https://docs.docker.com/engine/reference/commandline/create/) to know more |
+ *
+ * ### 3. Create selenoid container
+ * Simplest way to to this is enabling `autoCreate` and it will be taken care by codecept itself.
+ * **If you are using windows machine or if `autoCreate` does not work properly, create container manually**
+ *
+ * Run the following command to create one. To know more [refer here](https://aerokube.com/selenoid/latest/#_option_2_start_selenoid_container)
+ * ```
+ * docker create                                    \
+ * --name selenoid                                  \
+ * -p 4444:4444                                     \
+ * -v /var/run/docker.sock:/var/run/docker.sock     \
+ * -v `pwd`/:/etc/selenoid/:ro                      \
+ * -v `pwd`/output/video/:/opt/selenoid/video/      \
+ * -e OVERRIDE_VIDEO_OUTPUT_DIR=`pwd`/output/video/ \
+ * aerokube/selenoid:latest-release
+ * ```
+ */
+
+const selenoid = (config) => {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Block must not be padded by blank lines.](https://app.codacy.com/manual/DavertMik/CodeceptJS/pullRequest?prid=4798440)